core/base/mtx_io.cpp

```diff
@@ -845,7 +845,7 @@ matrix_data<ValueType, IndexType> read_binary_convert(std::istream& is,
         throw GKO_STREAM_ERROR(
             "cannot read into this format, would assign complex to real");
     }
-    matrix_data<ValueType, IndexType> result(gko::dim<2>{num_rows, num_cols});
+    matrix_data<ValueType, IndexType> result(gko::dim<2>{static_cast<size_type>(num_rows), static_cast<size_type>(num_cols)});
```

Comment:
I guess you compiled for x86 (32 bit) here?

Reply:
right